scripts/shop_import/create_prices_from_csv.py

```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+import time
+
+import openfoodfacts
+
+from scripts.utils import create_price, read_csv
+
+OPEN_PRICES_CREATE_PRICE_ENDPOINT = f'{os.environ.get("API_ENDPOINT")}/prices'
+OPEN_PRICES_TOKEN = os.environ.get("API_TOKEN")
+
+REQUIRED_ENV_PARAMS = [
+    # "FILEPATH"
+    # "DELIMITER" (optional)
+    "PRODUCT_CODE_FIELD",
+    "PRODUCT_NAME_FIELD",
+    "PRICE_FIELD",
+    "CURRENCY",
+    "LOCATION_OSM_ID",
+    "LOCATION_OSM_TYPE",
+    "DATE",
+    "PROOF_ID",
+    "API_ENDPOINT",
+    "API_TOKEN",
+    # DRY_RUN
+]
+
+
+def map_gdpr_price_list_to_open_prices(price_list, extra_data={}):
+    # map source fields to op fields
+    open_prices_price_list = list()
+    for price in price_list:
+        open_prices_price = dict()
+        # product_name
+        if os.environ.get("PRODUCT_NAME_FIELD"):
+            open_prices_price["product_name"] = price.get(
+                os.environ.get("PRODUCT_NAME_FIELD")
+            )
+        # product_code
+        open_prices_price["product_code"] = price.get(
+            os.environ.get("PRODUCT_CODE_FIELD")
+        )
+        # price
+        price_str = price.get(os.environ.get("PRICE_FIELD"))
+        open_prices_price["price"] = (
+            float(price_str.replace(",", ".")) if price_str else None
+        )
+        # print(open_prices_price)
+        open_prices_price_list.append({**open_prices_price, **extra_data})
+
+    return open_prices_price_list
+
+
+def filter_rules(op_price_list):
+    """
+    Rules to skip some prices (on code, name...)
+    """
+    op_price_list_filtered = list()
+
+    for op_price in op_price_list:
+        passes_test = True
+
+        if not op_price["product_code"]:
+            passes_test = False
+        elif not op_price["product_code"].isnumeric():
+            passes_test = False
+        elif len(op_price["product_code"]) < 6:
+            passes_test = False
+        elif not openfoodfacts.barcode.has_valid_check_digit(op_price["product_code"]):
```

Comment:
thanks to https://github.com/openfoodfacts/openfoodfacts-python/commit/6faadc1ea00b9ade6dfad50ba9cd5c261a4dcf8a